patroni/daemon.py

```diff
@@ -54,7 +54,11 @@ def run(self):
         while not self.received_sigterm:
             if self._received_sighup:
                 self._received_sighup = False
-                self.reload_config(True, self.config.reload_local_configuration())
+                if self.config.reload_local_configuration() or self.api.reload_local_certificate():
+                    reload = True
+                else:
+                    reload = False
+                self.reload_config(True, reload)
```

Comment:
This is not really the right place (abstract class, which is also used by patroni_raft_controller).
The `api.reload_conf()` should be called from the https://github.com/zalando/patroni/blob/master/patroni/__init__.py#L77
Condition would be: `if local or self.api.reload_local_certificate()`.